src/index.js

```diff
@@ -4,19 +4,19 @@ export const BEGIN = '@@optimist/BEGIN';
 export const COMMIT = '@@optimist/COMMIT';
 export const REVERT = '@@optimist/REVERT';
 
-export const ensureState = state => {
-  if (Map.isMap(state)) {
-    if (List.isList(state.get('history'))) {
-      return state.get('current');
-    }
-  }
-  return state;
-};
+// this marker is used to identify our wrapped state. The marker is designed to prevent false positives
+// that might occur if we checked for the existence of other properties. It is exported for testing.
+export const STATE_MARKER = '__redux-optimistic-ui';
```

Comment:
yeah, it's a tradeoff. you add a statemarker & you get a bloated state which isn't fun when sifting through the redux devtool developer. you're polluting the developer's space for the benefit of implementation.
but, if you use ducktyping, it's always possible to catch a false positive.
personally, i'd opt for the ducktyping. you could do something like:
`return Object.keys(state).length === 3 && state.hasOwnProperty('beforeState') && state.hasOwnProperty('history') && state.hasOwnProperty('current')`. It'd be almost impossible to have a collision then, and if there were one, the developer could change their shape.